packages/confirm-dialog/src/vaadin-confirm-dialog-mixin.js

```diff
@@ -0,0 +1,442 @@
+         * @type {boolean}
+         */
+        cancelButtonVisible: {
+          type: Boolean,
+          reflectToAttribute: true,
+          value: false,
+        },
+
+        /**
+         * Text displayed on cancel-button.
+         * This only affects the default button, custom slotted buttons will not be altered.
+         * @attr {string} cancel-text
+         * @type {string}
+         */
+        cancelText: {
+          type: String,
+          value: 'Cancel',
+        },
+
+        /**
+         * Theme for a cancel-button.
+         * This only affects the default button, custom slotted buttons will not be altered.
+         * @attr {string} cancel-theme
+         * @type {string}
+         */
+        cancelTheme: {
+          type: String,
+          value: 'tertiary',
+        },
+
+        /**
+         * A space-delimited list of CSS class names
+         * to set on the underlying overlay element.
+         *
+         * @attr {string} overlay-class
+         */
+        overlayClass: {
+          type: String,
+        },
+
+        /**
+         * A reference to the "Cancel" button which will be teleported to the overlay.
+         * @private
+         */
+        _cancelButton: {
+          type: Object,
+        },
+
+        /**
+         * A reference to the "Confirm" button which will be teleported to the overlay.
+         * @private
+         */
+        _confirmButton: {
+          type: Object,
+        },
+
+        /**
+         * A reference to the "header" node which will be teleported to the overlay.
+         * @private
+         */
+        _headerNode: {
+          type: Object,
+        },
+
+        /**
+         * A list of message nodes which will be placed in the overlay default slot.
+         * @private
+         */
+        _messageNodes: {
+          type: Array,
+          value: () => [],
+        },
+
+        /**
+         * A reference to the "Reject" button which will be teleported to the overlay.
+         * @private
+         */
+        _rejectButton: {
+          type: Object,
+        },
+
+        /**
+         * Height to be set on the overlay content.
+         * @protected
+         */
+        _contentHeight: {
+          type: String,
+        },
+
+        /**
+         * Width to be set on the overlay content.
+         * @protected
+         */
+        _contentWidth: {
+          type: String,
+        },
+      };
+    }
+
+    static get observers() {
+      return [
+        '__updateConfirmButton(_confirmButton, confirmText, confirmTheme)',
+        '__updateCancelButton(_cancelButton, cancelText, cancelTheme, cancelButtonVisible)',
+        '__updateHeaderNode(_headerNode, header)',
+        '__updateMessageNodes(_messageNodes, message)',
+        '__updateRejectButton(_rejectButton, rejectText, rejectTheme, rejectButtonVisible)',
+        '__accessibleDescriptionRefChanged(_overlayElement, _messageNodes, accessibleDescriptionRef)',
```

Comment:
This observer was changed to also include `_messageNodes` to handle `aria-describedby` correctly when the overlay is initialized, rather than doing that in `this._messageController` initializer callback.